packages/flame/lib/src/components/mixins/has_paint.dart

```diff
@@ -0,0 +1,81 @@
+import 'dart:ui';
+
+import '../../../components.dart';
+import '../../palette.dart';
+
+const _kMain = 'main';
+
+/// Adds a collection of paints to a component
+///
+/// Component will always have a main Paint that can be accessed
+/// by the [paint] getter/setter and other paints can be manipulated/accessed
+/// using [getPaint], [setPaint] and [deletePaint]
+mixin HasPaint on BaseComponent {
+  final Map<String, Paint> _paints = {
```

Comment:
do we have any comp that has more than one paint? I found this map quite confusing to understand

Reply:
Not yet, but with this we could allow users to have custom components with more than one paints. Initially I had a hard time understanding why to have a collection of paint but @spydon made some fair points about this, there were a discussion about this point on this PR, take a look on it for more info

Reply:
hum, I see. I thought it was adding a lot of complexity, but I can see the use case for multiple paints. I have a different proposal though, because I dislike having arbitrary "strings" as keys, as that is not type safe, and subject to errors. we could use enums as keys. for example, see very bare-bones implementation (just to explain what I mean):

```dart
class Component {}

// generic HasPaints mixin has all the functionality and works with one or many paints
mixin HasPaints<T> on Component {
  Map<T, String> values = {};
}

// has paint always forced to have one paint
enum DefaultKey {  Main }
typedef HasPaint<T> = HasPaints<DefaultKey>;

class SimpleComponent extends Component with HasPaint {
  SimpleComponent() {
    values[DefaultKey.Main] = "bar";
    // this will not compile
    // values["Mani"] = "";
  }
}

// for complex components, you are forced to be type safe with your keys
enum EnumA { 
  Foo, Bar
}


class ComplexComponent extends Component with HasPaints<EnumA> {
  ComplexComponent() {
    values[EnumA.Foo] = "bar";
    // this will not compile
    // values[DefaultKey.Main] = "";
  }
}
```

That way whoever uses `HasPaint` is forced to have a single paint and ignores all the complexity, and however wats to use `HasPaints` has nothing to do with "main" and can create an enum to represent their paints the way they like, without magic strings.

Reply:
Interesting idea, I didn't liked the need for the user to need to initialize the main paint on the component, user will need to know the existence of the defaultkey enum, etc.

I will think if I can think on a middle ground solution for this

Reply:
ah that was just an example to prove it's type safe, you can have easy to use methods defined on the mixins to faciltiate access

Reply:
true, but I think I can think on a good way for implementing that, will implement it tonight :D